ext/random/engine_xoshiro256starstar.c

```diff
@@ -205,10 +205,12 @@ PHP_METHOD(Random_Engine_Xoshiro256StarStar, __construct)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (seed_is_null) {
-		if (php_random_bytes_throw(&state->state, 32) == FAILURE) {
-			zend_throw_exception(random_ce_Random_RandomException, "Failed to generate a random seed", 0);
-			RETURN_THROWS();
-		}
+		do {
+			if (php_random_bytes_throw(&state->state, 32) == FAILURE) {
+				zend_throw_exception(random_ce_Random_RandomException, "Failed to generate a random seed", 0);
+				RETURN_THROWS();
+			}
+		} while (UNEXPECTED(state->state[0] == 0 && state->state[1] == 0 && state->state[2] == 0 && state->state[3] == 0));
```

Comment:
I'm not too happy about hard-coding the number of elements here, but can't come up with a more elegant solution, so I guess it's okay.

Reply:
Yeah, it also looked pretty verbose to me. But on the other hand it's not like the number of elements is ever going to change. Any kind of loop would make the code more complex for an extraordinarily uncommon case.

I thought about ORing all the elements together, but that would be less clear (interestingly gcc already optimizes the condition to `(state->state[0] | state->state[1] | state->state[2] | state->state[3]) == 0`).